sklearn/model_selection/_validation.py

```diff
@@ -468,9 +468,18 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
         # _score will return dict if is_multimetric is True
         test_scores = _score(estimator, X_test, y_test, scorer, is_multimetric)
         score_time = time.time() - start_time - fit_time
-        if return_train_score:
+        if return_train_score is True:
+            train_scores = _score(estimator, X_train, y_train, scorer,
+                                  is_multimetric)
+        if return_train_score is 'warn':
             train_scores = _score(estimator, X_train, y_train, scorer,
                                   is_multimetric)
+            score_train_time = time.time() - start_time - fit_time - score_time
+            if (score_train_time >= 0.1*fit_time and
+               time.time() - start_time) > 5:
```

Comment:
Parentheses in a strange place here...

And can't compare absolute time to 5 in a single fit. It will often only be a substantial amount of time over a large number of fits, as in the raising issue.

Reply:
How about adding a new variable alongside `return_train_score` in BaseSearchCV, containing the time of the first call to GridSearchCV. Also we need to pass that time variable to the _fit_and_score function.